Add security advisor role

dmcgowan opened the pull request:
Part of an effort to involve more contributors to the project in a variety of roles.

I want to be able to keep the reviewer role focused on PR and issues while continuing to expand the number of reviewers. At the same time we need review of specific security issues by a smaller group who may have less background in the code of the project, but rather the impact of the issue.

9 maintainer LGTM required (2/3)

- [x] @AkihiroSuda
- [x] @crosbymichael
- [x] @dmcgowan
- [x] @estesp 
- [ ] @jterry75
- [x] @mlaventure
- [x] @stevvooe
- [ ] @dchen1107
- [ ] @Random-Liu
- [x] @mikebrow
- [ ] @yujuhong
- [x] @fuweid
- [x] @mxpv

mikebrow commented on GOVERNANCE.md:
```diff
@@ -61,6 +61,31 @@ maintainer has already discussed the candidacy with the candidate and a
+individuals trusted by maintainers and representing significant users of the
+project.
+
+Security advisors are part of the organization without write access, but read
```
s/but read/but with read/

crosbymichael commented on SECURITY_ADVISORS:
```diff
@@ -0,0 +1,6 @@
+# See GOVERNANCE.md for description of role
+#
+# SECURITY ADVISORS
+# GitHub ID, Name, Email address
```
Should we have a column for who these advisors may represent?  i.e. a company?

fuweid replied:
Need the member to turn on 2FA for GitHub.

fuweid replied:
I think we should require the member to turn on 2FA for GitHub account, for security reason

thaJeztah replied:
Good question; guess it depends; if these people represent a company does that also (implicitly) means that if they would move companies, that another person should take their place?

dmcgowan replied:
I was thinking about this. The email address here should really use a company email. This email will get linked to `security@containerd.io`, so moving jobs should mean updating here. We can leave it up to maintainer discretion. We should avoid adding other mailing lists here though since the goal is to get an individual's advice.

Adding company is good though so those within a company may know who to contact related to containerd security.

mikebrow replied:
> if these people represent a company does that also (implicitly) means that if they would move companies, that another person should take their place?

> We can leave it up to maintainer discretion.

+1

estesp commented:
SGTM. The only open question in my mind is whether there is yet another "receive only" role for end user/vendors who don't feel they have the ability/interest to be more actively involved in security issues and only want to be part of a "disclose" list for a heads up on potential CVEs and mitigations for their production systems? Or do we want to make sure that all end users end up with at least one "security advisor" here? I haven't formed a strong opinion either way, but I wonder if we need to call that out specifically if we aren't going to have another (mailing list?) avenue for security disclosure awareness and announces.

dmcgowan commented:
@estesp we already have `containerd-security-announce@lists.cncf.io` but don't actually have a way to get added to this list. I think step 2 of this is to add `SECURITY.md` which mentions this security list and the process for getting added to it. That list will be different as it is likely other mailing lists and not necessarily a list we want to be public. I imagine the process will be to request access via `security@containerd.io`

dmcgowan commented:
> Is this type of roll going to replace mailing lists for security issues as we can handle most of the notification and embargo via GitHub now?

Not sure if GitHub has an equivalent of the announce list, which gives a slightly larger set of organizations some details of an issue as well as a fix. If the announce list is unnecessary, that is fine, but I have a feeling some organizations will want email notification via their security list.

dmcgowan commented:
Updated and marked ready for review. Added "Company", but "Independent" is always a possible value if that makes sense or necessary for privacy reasons.

AkihiroSuda commented on GOVERNANCE.md:
```diff
@@ -61,6 +61,31 @@ maintainer has already discussed the candidacy with the candidate and a
+project, there is no expectation that they are well known by a majority of
+maintainers. Approving the pull request requires a two LGTM threshold of
+maintainers.
+
```
An inactive maintainer/reviewer is expected to be demoted/removed. What is the criteria for removing security advisors?

dmcgowan replied:
Added section, same sponsoring maintainer + 2 LGTM to remove or 33%

thaJeztah commented on GOVERNANCE.md:
```diff
@@ -61,6 +61,31 @@ maintainer has already discussed the candidacy with the candidate and a
+
+Security advisors are part of the organization without write access, but read
+access to security disclosures and advisories before becoming public. There
+is no expectation of advisors to be become reviewers or participate in issue
```
```suggestion
is no expectation of advisors to become reviewers or participate in issue
```

crosbymichael commented:
LGTM

thaJeztah commented:
@stevvooe @jterry75 @dchen1107 @Random-Liu @yujuhong ptal

stevvooe commented on GOVERNANCE.md:
```diff
@@ -61,6 +61,35 @@ maintainer has already discussed the candidacy with the candidate and a
+Security advisors are part of the organization without write access, but with
+read access to security disclosures and advisories before becoming public. There
+is no expectation of advisors to become reviewers or participate in issue
+triage and code review. Security advisors help maintain the integrity of the
```
Should we add something here about developing threat models?

dmcgowan replied:
Thread model development is a fairly specific task, I kind of left the role as more high level purpose. Maybe mentioning in the first sentence under "help classify" and "advise"

thaJeztah commented:
Looks like we have 9 LGTM's, do we want to wait for  @jterry75 @dchen1107 @Random-Liu @yujuhong ?

dmcgowan commented:
Usually we give it a week for discussion, I'll go ahead and merge.

We discussed having a `SECURITY.md` as a follow up here as well with more specific details about the security process.
